src/python/dxpy/scripts/dx.py

```diff
@@ -3785,6 +3785,155 @@ def publish(args):
     except:
         err_exit()
 
+def archive(args):
+    def send_archive_request(target_project, request_input, request_func):
+        api_errors = [InvalidState, ResourceNotFound, PermissionDenied]
+        try:
+            res = request_func(target_project, request_input)
+        except Exception as e:            
+            eprint("Failed request: {}".format(request_input))
+            if type(e) in api_errors:
+                eprint("     API error: {}. {}".format(e.name, e.msg))
+            else: 
+                eprint("     Unexpected error: {}".format(format_exception(e)))
+            
+            err_exit("Failed request: {}. {}".format(request_input, format_exception(e)), code=3)
+        return res              
+
+    def get_valid_archival_input(args, target_files, target_folder, target_project):
+        request_input = {}
+        if target_files and target_folder:
+            err_exit('Expecting either a single folder or a list files for each API request', code=3)
+        elif target_files: 
+            request_input = {"files": target_files}
+        elif target_folder:
+            request_input = {"folder": target_folder, "recurse":args.recurse}
+        else:
+            err_exit("No input file/folder is found in project {}".format(target_project), 
+                            code=3)
+        return request_input
+
+    def get_archival_paths(args):
+        target_project = None
+        target_folder = None
+        target_files = []
+        
+        paths = [split_unescaped(':', path, include_empty_strings=True) for path in args.path]
+
+        for path in paths:
+            try: 
+                if len(path) > 2: # ignore invalid path
+                    continue
+
+                # if the input path is in the following format:
+                #   file-xxxx
+                #   /folderpath/filename or /folderpath/
+                # set the target project to be the current one
+                elif len(path) == 1:
+                    if not target_project:
+                        target_project = dxpy.WORKSPACE_ID 
```

Comment:
`dxpy.WORKSPACE_ID` should only exist if the call is made from a running job (http://autodoc.dnanexus.com/bindings/python/current/dxpy.html).  In general, the file could not exist in the current project (`dxpy.PROJECT_CONTEXT_ID`).  My feeling is it may be best in this case to require a project ID and error out in case one isn't provided.  It's not the 'classic' way we have done things, but it is a safer, best practice that we can begin enforcing more.  We have a design project planned this quarter for how we can do this more hoistically.  @emiloslavsky interest.

Reply:
Fixed.

Reply:
@emiloslavsky can you help us make a product-level decision here based on the convo above?  Currently Yuxin defaults to the project context (which makes sense under certain assumptions), but we have an alternative approach that I described above to error out if it's not provided.  There are certainly pros and cons to both and just want to make sure you have the final say there.

Reply:
Actually, nevermind I think I see what you're doing @YuxinShi0423 but would like to clarify with you.  I'll try to make it to the standup tomorrow to discuss more on the tail end of the standup.   Just want to make sure we're on the same page. Cheers!